src/components/MediaDevices/VideoCheck.tsx

```diff
@@ -0,0 +1,187 @@
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+import {
+  useVideoTrack,
+  useDevices,
+  useLocalParticipant,
+} from '@daily-co/daily-react';
+import { Link } from 'react-router-dom';
+
+import { Card } from '../shared/Card/Card.tsx';
+import { TroubleShooting } from '../shared/TroubleShooting/TroubleShooting.tsx';
+import { Loader } from '../shared/Loader/Loader.tsx';
+
+export const VideoCheck: React.FC = () => {
+  const localParticipantId = useLocalParticipant();
+
+  const videoTrack = useVideoTrack(
+    localParticipantId ? localParticipantId.session_id : '',
+  );
+  const videoElement = useRef<HTMLVideoElement>(null);
+  const [showTroubleshooting, setShowTroubleshooting] = useState(false);
+
+  const { cameras, setCamera, hasCamError, camState } = useDevices();
+
+  useEffect(() => {
+    if (!videoTrack.persistentTrack) return;
+    if (videoElement.current) {
+      videoElement.current.srcObject =
+        videoTrack.persistentTrack &&
```

Comment:
I think this check is not needed, because we already safe-guarded at the beginning of the effect function.